### PR TITLE
Add character icons flipped

### PR DESCRIPTION
**Pull request by ArthurJordao**
# :wrench: Modifying a component

## Context

We need the character icons flipped for the short response guided drafts. [Example on figma](https://www.figma.com/file/OAu5rysVeG2QsKyOlN9YXX/Short-Response-GD%3A-Student-XP?type=design&node-id=2763-148078&mode=design&t=hM46b6Ik3ieLchIe-0).

## :framed_picture: What does this change look like?

- [x] Follow the the [Guidelines for an optimal design ping](https://paper.dropbox.com/doc/Guidelines-for-Sharing-User-Facing-Changes-with-Design--BpL8hpJLMugy6033aT5m0JdaAg-bdKGQtYH9qO9I00hUkA6k) to add helpful screenshots/videos for design.
- [x] @ mention the design team on this PR to request approval on your changes. A designer will "thumbs up" react to your PR if they approve it, or will leave comments if it's not quite approved yet.
Screenshot 2023-08-11 at 17 26 29

## Component completion checklist

- [x] I've gone through the relevant sections of the [Development Accessibility guide](https://paper.dropbox.com/doc/Accessibility-guide-4-Development--BiIVdijSaoijjOuhz3iTCJJ1Ag-rGoHpC91pFg3zTrYpvOCQ) with the changes I made to this component in mind
- [x] Changes are clearly documented
    - [x] Component docs include a changelog
    - [x] Any new exposed functions or properties have docs
- [x] Changes extend to the Component Catalog
    - [x] The Component Catalog is updated to use the newest version, if appropriate
    - [x] The Component Catalog example version number is updated, if appropriate
    - [x] Any new customizations are available from the Component Catalog
    - [x] The component example still has:
        - an accurate preview
        - valid sample code
        - correct keyboard behavior
        - correct and comprehensive guidance around how to use the component
- [x] Changes to the component are tested/the new version of the component is tested
- [x] Component API follows standard patterns in noredink-ui
    - e.g., as a dev, I can conveniently add an `nriDescription`
    - and adding a new feature to the component will _not_ require major API changes to the component
- [x] If this is a new major version of the component, our team has stories created to upgrade all instances of the old component. Here are links to the stories:
  - add your story links here OR just write this is not a new major version

### CODE REVIEW

File: src/Nri/Ui/CharacterIcon/V2.elm [hunk 1]
```diff
@@ -1159,30 +1161,34 @@ renderHeadshotFrame :
     , clippedSvg : Svg.Svg Never
     , secondaryPath : String
     , secondaryColor : Css.Color
+    , circleStyles : List Css.Style
+    , styles : List Css.Style
```
Comment (BrianHicks) [inline src/Nri/Ui/CharacterIcon/V2.elm:1165]: we never use `styles` in this record any more. What if we were explicit about this by calling this `circleClipStyles` and then explicitly passing some empty lists? We got lucky that leaving the styles out of that element worked this time, but next time we're styling something it'll be weird for styles not to be applied to some arbitrary element.

If that doesn't sound good to you, could you add a note so we know where styles aren't being applied so someone could fix it up when they need it in the future?

**Review by BrianHicks — CHANGES_REQUESTED** [review]
I'm glad this approach ended up working! I just have a little thought about maintainability.

File: src/Nri/Ui/CharacterIcon/V2.elm (hunk 1 above)
Comment (ArthurJordao) [reply src/Nri/Ui/CharacterIcon/V2.elm:1165]: Actually I think we don't need those anymore, we just need the circle styles. I don't think we will need them.

**Review by BrianHicks — APPROVED** [review]
`flipStyles` makes this much clearer. Thanks for that refactor!

flip-cooking-gif-by-bluesbear

**Review by tesk9 — COMMENTED** [review]
I'm curious about adding new icons rather than using: 

```
CharacterIcon.lindyHeadshot
        |> Svg.withCss [ Css.transforms [ Css.scaleX -1 ] ]
```

in the context where you need it.

Are you expecting to have the same character appear multiple times on the same view at the same time, facing different directions?

(To be clear -- I'm not opposed to adding these icons. I'm just wondering if flipping the character with CSS right where you're using it might be easier)

**Comment by BrianHicks** [comment]
That'd be fine for Red and Sal, but we want Lindy's scarf to face one way and their face the other.

**Comment by tesk9** [comment]
Oh dang! Now I see it -- fabulous attention to detail! 😍 

Okay that makes me wonder about the names -- what would you think about `lindyHeadshotFacingRight` or something like that instead of `lindyHeadshotFlipped`? "Flipped" implies a simple transformation to me, personally.

**Comment by ArthurJordao** [comment]
@tesk9 makes sense! I will do the change here 🧑‍🏭

**Comment by tesk9** [comment]
One small other thing -- looks like Percy has caught an ear regression:
image

(honestly, I kinda dig it -- very batty or something)

**Comment by ArthurJordao** [comment]
@tesk9 yeah, I am on it, for some reason when I changed the ids this got broken

**Comment by staceyadams** [comment]
Adds another option to the internal debate -- is Red a red panda, fox, or a bat?

**Review by staceyadams — APPROVED** [review]

**Comment by staceyadams** [comment]
(just to be super clear -- my approval is based on the first screenshot, not the batty version of Red 😄)

**Comment by ArthurJordao** [comment]
@staceyadams hahaha it is fixed now. Thanks 😄

**Review by tesk9 — APPROVED** [review]
Thank you! As always, tremendously cute. Thanks for working through the duplicate id issues in particular! 🙏 

Heads up on timing: please don't do a release until https://github.com/NoRedInk/noredink-ui/pull/1459 merges, so that we avoid an accidental major version bump. Feel free to merge that PR and do a release if you're blocked tho! If you don't do a release, I expect I'll do one Monday or Tuesday of next week, as we've got a lot of changes ready to go!

**Comment by ArthurJordao** [comment]
Thank you for the review and the heads-up @tesk9

**Pull request merged**